components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -551,6 +580,83 @@ class NewTabPage extends React.Component<Props, State> {
     })
   }
 
+  getCryptoDotComTickerInfo = (asset: string) => {
+    return new Promise((resolve: Function) => {
+      chrome.cryptoDotCom.getTickerInfo(`${asset}_USDT`, (resp: any) => {
+        resolve({ [asset]: resp })
+      })
+    })
+  }
+
+  getCryptoDotComAssetRankings = () => {
+    return new Promise((resolve: Function) => {
+      chrome.cryptoDotCom.getAssetRankings((resp: any) => {
+        resolve(resp)
+      })
+    })
+  }
+
+  getCryptoDotComChartData = (asset: string) => {
+    return new Promise((resolve: Function) => {
+      chrome.cryptoDotCom.getChartData(`${asset}_USDT`, (resp: any) => {
+        resolve({ [asset]: resp })
+      })
+    })
+  }
+
+  getCryptoDotComSupportedPairs = () => {
+    return new Promise((resolve: Function) => {
+      chrome.cryptoDotCom.getSupportedPairs((resp: any) => {
+        resolve(resp)
+      })
+    })
+  }
+
+  fetchCryptoDotComTickerPrices = async (assets: string[]) => {
+    const assetReqs = assets.map(asset => this.getCryptoDotComTickerInfo(asset))
+    const assetResps = await Promise.all(assetReqs).then((resps: object[]) => resps)
+    const assetPrices = assetResps.reduce((all, current) => ({ ...current, ...all }), {})
+    this.props.actions.setCryptoDotComTickerPrices(assetPrices)
+  }
+
+  fetchCryptoDotComCharts = async (assets: string[]) => {
+    const chartReqs = assets.map(asset => this.getCryptoDotComChartData(asset))
+    const chartResps = await Promise.all(chartReqs).then((resps: object[]) => resps)
+    const charts = chartResps.reduce((all, current) => ({ ...current, ...all }), {})
+    this.props.actions.setCryptoDotComCharts(charts)
+  }
+
+  fetchCryptoDotComLosersGainers = async () => {
+    const losersGainers = await this.getCryptoDotComAssetRankings().then((resp: any) => resp)
+    this.props.actions.setCryptoDotComLosersGainers(losersGainers)
+  }
+
+  fetchCryptoDotComSupportedPairs = async () => {
+    const pairs = await this.getCryptoDotComSupportedPairs().then((resp: any) => resp)
+    this.props.actions.setCryptoDotComSupportedPairs(pairs)
+  }
+
+  cryptoDotComUpdateActions = async () => {
+    const assets = Object.keys(this.props.newTabData.cryptoDotComState.tickerPrices)
+    return Promise.all([
+      this.fetchCryptoDotComTickerPrices(assets),
+      this.fetchCryptoDotComLosersGainers(),
+      this.fetchCryptoDotComCharts(assets)
```

Comment:
This causes 3 separate redux actions to be fired. Only case I could see we want this is if some data comes back a long time before other data and we want to render whatever comes first. OTherwise if it all comes back at the same time it would be faster to get all the data and then fire off one action. Also the firing of this could all be handled in redux and not this react component.